File: web/cursor_interop.js
// Set with custom image and fallback
window.flutterSetCursor = function(cursorUrl) {
  try {
    const img = new Image();
    img.src = cursorUrl;

    img.onload = function() {
      const xHotspot = Math.floor(img.width / 2);
      const yHotspot = Math.floor(img.height / 2);
      document.body.style.cursor = `url('${cursorUrl}') ${xHotspot} ${yHotspot}, auto`;
    };

    img.onerror = function() {
      console.warn('Cursor image failed to load, using fallback');
      document.body.style.cursor = 'pointer';
    };
  } catch (e) {
    console.error('Cursor error:', e);
    document.body.style.cursor = 'auto';
  }
};

// Reset to default
window.flutterResetCursor = function() {
  document.body.style.cursor = 'auto';
};

File: cursor_interop.dart
// import 'dart:js/js_wasm.dart';
//
// @JS('flutterSetCursor')
// external void setCustomCursor(String cursorUrl);
//
// @JS('flutterResetCursor')
// external void resetCursor();
//
// class WebCursor {
//   static void setCursor(String imagePath, {String fallback = 'pointer'}) {
//     try {
//       if (imagePath == 'auto') {
//         resetCursor();
//       } else {
//         setCustomCursor(imagePath);
//       }
//     } catch (e) {
//       _setBasicCursor(fallback);
//     }
//   }
//
//   static void reset() {
//     try {
//       resetCursor();
//     } catch (e) {
//       _setBasicCursor('auto');
//     }
//   }
//
//   @JS('document.body.style.cursor')
//   external static set _setBasicCursor(String cursorType);
// }